--- w32api.orig/GL/glu.h	2014-04-18 22:30:50.186158900 -0600
+++ w32api/GL/glu.h	2014-04-18 22:42:59.095926900 -0600
@@ -108,8 +108,12 @@
 typedef void (CALLBACK *GLUtessCombineDataProc)(GLdouble[3],void*[4],GLfloat[4],void**,void*);
 typedef void (CALLBACK *GLUnurbsErrorProc)(GLenum);
 
+#define GLU_EXT_object_space_tess          1
+#define GLU_EXT_nurbs_tessellator          1
+
 #define GLU_VERSION_1_1 1
 #define GLU_VERSION_1_2 1
+#define GLU_VERSION_1_3                    1
 
 #define GLU_INVALID_ENUM 100900
 #define GLU_INVALID_VALUE 100901
@@ -183,7 +187,17 @@
 #define GLU_SAMPLING_METHOD 100205
 #define GLU_U_STEP 100206
 #define GLU_V_STEP 100207
-
+#define GLU_NURBS_MODE                     100160
+#define GLU_NURBS_MODE_EXT                 100160
+#define GLU_NURBS_TESSELLATOR              100161
+#define GLU_NURBS_TESSELLATOR_EXT          100161
+#define GLU_NURBS_RENDERER                 100162
+#define GLU_NURBS_RENDERER_EXT             100162
+
+#define GLU_OBJECT_PARAMETRIC_ERROR        100208
+#define GLU_OBJECT_PARAMETRIC_ERROR_EXT    100208
+#define GLU_OBJECT_PATH_LENGTH             100209
+#define GLU_OBJECT_PATH_LENGTH_EXT         100209
 #define GLU_PATH_LENGTH 100215
 #define GLU_PARAMETRIC_ERROR 100216
 #define GLU_DOMAIN_DISTANCE 100217
@@ -194,6 +208,33 @@
 #define GLU_OUTLINE_POLYGON 100240
 #define GLU_OUTLINE_PATCH 100241
 
+#define GLU_NURBS_ERROR                    100103
+#define GLU_ERROR                          100103
+#define GLU_NURBS_BEGIN                    100164
+#define GLU_NURBS_BEGIN_EXT                100164
+#define GLU_NURBS_VERTEX                   100165
+#define GLU_NURBS_VERTEX_EXT               100165
+#define GLU_NURBS_NORMAL                   100166
+#define GLU_NURBS_NORMAL_EXT               100166
+#define GLU_NURBS_COLOR                    100167
+#define GLU_NURBS_COLOR_EXT                100167
+#define GLU_NURBS_TEXTURE_COORD            100168
+#define GLU_NURBS_TEX_COORD_EXT            100168
+#define GLU_NURBS_END                      100169
+#define GLU_NURBS_END_EXT                  100169
+#define GLU_NURBS_BEGIN_DATA               100170
+#define GLU_NURBS_BEGIN_DATA_EXT           100170
+#define GLU_NURBS_VERTEX_DATA              100171
+#define GLU_NURBS_VERTEX_DATA_EXT          100171
+#define GLU_NURBS_NORMAL_DATA              100172
+#define GLU_NURBS_NORMAL_DATA_EXT          100172
+#define GLU_NURBS_COLOR_DATA               100173
+#define GLU_NURBS_COLOR_DATA_EXT           100173
+#define GLU_NURBS_TEXTURE_COORD_DATA       100174
+#define GLU_NURBS_TEX_COORD_DATA_EXT       100174
+#define GLU_NURBS_END_DATA                 100175
+#define GLU_NURBS_END_DATA_EXT             100175
+
 #define GLU_NURBS_ERROR1 100251
 #define GLU_NURBS_ERROR2 100252
 #define GLU_NURBS_ERROR3 100253
@@ -248,6 +289,13 @@
 #define GLU_ERROR GLU_TESS_ERROR
 #define GLU_EDGE_FLAG GLU_TESS_EDGE_FLAG
 
+/* Internal convenience typedefs */
+#ifdef __cplusplus
+typedef void (APIENTRY *_GLUfuncptr)();
+#else
+typedef void (APIENTRY *_GLUfuncptr)(GLvoid);
+#endif
+ 
 #ifdef __cplusplus
 }
 #endif
