diff -r -u freeglut-2.8.1/src/freeglut_internal.h freeglut-2.8.1J/src/freeglut_internal.h
--- freeglut-2.8.1/src/freeglut_internal.h	2012-11-19 06:14:16.000000000 -0700
+++ freeglut-2.8.1J/src/freeglut_internal.h	2018-01-05 11:38:25.214161500 -0700
@@ -32,6 +32,9 @@
 #    include "config.h"
 #endif
 
+#define GCL_HCURSOR (-12)
+#define _snprintf snprintf
+
 /* XXX Update these for each release! */
 #define  VERSION_MAJOR 2
 #define  VERSION_MINOR 8
