diff -ru gc-7.6.2.orig/include/config.h.in gc-7.6.2/include/config.h.in
--- gc-7.6.2.orig/include/config.h.in	2017-12-23 00:48:37.000000000 -0800
+++ gc-7.6.2/include/config.h.in	2018-01-12 13:42:30.026704300 -0800
@@ -100,19 +100,27 @@
 #undef GC_USE_DLOPEN_WRAP
 
 /* The major version number of this GC release. */
+#ifndef GC_VERSION_MAJOR
 #undef GC_VERSION_MAJOR
+#endif
 
 /* The micro version number of this GC release. */
+#ifndef GC_VERSION_MICRO
 #undef GC_VERSION_MICRO
+#endif
 
 /* The minor version number of this GC release. */
+#ifndef GC_VERSION_MINOR
 #undef GC_VERSION_MINOR
+#endif
 
 /* Define to support pthreads-win32 or winpthreads. */
 #undef GC_WIN32_PTHREADS
 
 /* Define to support Win32 threads. */
+#ifndef GC_WIN32_THREADS
 #undef GC_WIN32_THREADS
+#endif
 
 /* Define to install pthread_atfork() handlers by default. */
 #undef HANDLE_FORK
@@ -196,22 +204,32 @@
 #undef PACKAGE
 
 /* Define to the address where bug reports for this package should be sent. */
+#ifndef PACKAGE_BUGREPORT
 #undef PACKAGE_BUGREPORT
+#endif
 
 /* Define to the full name of this package. */
+#ifndef PACKAGE_NAME
 #undef PACKAGE_NAME
+#endif
 
 /* Define to the full name and version of this package. */
+#ifndef PACKAGE_STRING
 #undef PACKAGE_STRING
+#endif
 
 /* Define to the one symbol short name of this package. */
+#ifndef PACKAGE_TARNAME
 #undef PACKAGE_TARNAME
+#endif
 
 /* Define to the home page for this package. */
 #undef PACKAGE_URL
 
 /* Define to the version of this package. */
+#ifndef PACKAGE_VERSION
 #undef PACKAGE_VERSION
+#endif
 
 /* Define to enable parallel marking. */
 #undef PARALLEL_MARK
diff -ru gc-7.6.2.orig/include/gc_config_macros.h gc-7.6.2/include/gc_config_macros.h
--- gc-7.6.2.orig/include/gc_config_macros.h	2017-12-23 00:46:29.000000000 -0800
+++ gc-7.6.2/include/gc_config_macros.h	2018-01-12 13:44:29.838482500 -0800
@@ -48,7 +48,9 @@
 # define GC_LINUX_THREADS
 #endif
 #if defined(WIN32_THREADS)
+# ifndef GC_WIN32_THREADS
 # define GC_WIN32_THREADS
+# endif
 #endif
 #if defined(RTEMS_THREADS)
 # define GC_RTEMS_PTHREADS
