diff -ru freeglut-2.6.0/configure freeglut-2.6.0J/configure
--- freeglut-2.6.0/configure	2009-04-20 20:53:07.000000000 -0700
+++ freeglut-2.6.0J/configure	2009-06-22 06:49:17.640625000 -0700
@@ -21289,7 +21289,7 @@
   LIBXXF86VM=
   LIBXI=
   VERSION_INFO=
-  EXPORT_FLAGS="-DFREEGLUT_EXPORTS"
+  EXPORT_FLAGS="-DFREEGLUT_EXPORTS -D_WIN32"
 else
   GL_LIBS="-lGL -lXext -lX11"
   { echo "$as_me:$LINENO: checking for XF86VidModeSwitchToMode in -lXxf86vm" >&5
