diff -ru gv-3.6.1/src/Ghostview.c gv-3.6.1J/src/Ghostview.c
--- gv-3.6.1/src/Ghostview.c	2004-12-07 16:21:30.000000000 -0700
+++ gv-3.6.1J/src/Ghostview.c	2005-02-26 16:29:21.127613552 -0700
@@ -1428,12 +1428,6 @@
     if (gvw->ghostview.safer) 
       {
 	argv[argc++] = "-dSAFER";
-
-	/* PDF & DELAYSAFER hack */
-	if (pdf_delaysafer_hack) 
-	  {
-	    argv[argc++] = "-dDELAYSAFER";
-	  }
       }
 
     if (gvw->ghostview.arguments) {
@@ -1563,8 +1557,6 @@
     if (device)    GV_XtFree(device);
     gvw->ghostview.background_cleared=0;
 
-    pdf_delaysafer_hack = 0;
-
     ENDMESSAGE(StartInterpreter)
 }
 
diff -ru gv-3.6.1/src/main.c gv-3.6.1J/src/main.c
--- gv-3.6.1/src/main.c	2004-12-06 16:36:57.000000000 -0700
+++ gv-3.6.1J/src/main.c	2005-02-26 16:28:33.426865168 -0700
@@ -325,7 +325,6 @@
 /*###  initializing global variables ####################################*/
 
     INFMESSAGE(initializing global variables)
-    pdf_delaysafer_hack = 0;
     gv_scroll_mode = SCROLL_MODE_NONE;
     gv_class = GV_CLASS;
     gv_pending_page_request=NO_CURRENT_PAGE;
diff -ru gv-3.6.1/src/main_globals.h gv-3.6.1J/src/main_globals.h
--- gv-3.6.1/src/main_globals.h	2004-12-06 05:52:10.000000000 -0700
+++ gv-3.6.1J/src/main_globals.h	2005-02-26 16:28:33.427865016 -0700
@@ -41,8 +41,6 @@
 #  endif
 #endif
 
-EXTERN int pdf_delaysafer_hack;
-
 EXTERN Bool			show_date;
 EXTERN Bool			show_locator;
 EXTERN Bool			show_title;
diff -ru gv-3.6.1/src/misc.c gv-3.6.1J/src/misc.c
--- gv-3.6.1/src/misc.c	2004-11-11 05:21:05.000000000 -0700
+++ gv-3.6.1J/src/misc.c	2005-02-26 16:28:33.429864712 -0700
@@ -413,6 +413,14 @@
     if (!b && file_fileIsNotUseful(p)) sprintf(p,"%s.ps",name);  else b = True;
     if (!b && file_fileIsNotUseful(p)) sprintf(p,"%s.pdf",name); else b = True;
     if (!b)                            strcpy(p,name);
+    char *dir=file_getDirOfPath(p);
+    if(strcmp(dir,p)) {
+      chdir(dir);
+      // Strip off directory from p to satisfy GS 8.00 security change
+      char *q=p;
+      while(*dir && *dir == *q) {dir++; q++;}
+      strcpy(p,q);
+    }
   }
   name = p;
   INFSMESSAGE(trying to open,name)
diff -ru gv-3.6.1/src/ps.c gv-3.6.1J/src/ps.c
--- gv-3.6.1/src/ps.c	2004-12-07 16:55:59.000000000 -0700
+++ gv-3.6.1J/src/ps.c	2005-02-26 16:28:33.432864256 -0700
@@ -63,8 +63,6 @@
 
 #include "config.h"
 
-extern int pdf_delaysafer_hack;
-
 #include <stdio.h>
 #include <stdlib.h>
 
@@ -532,9 +530,6 @@
       section_len = line_len;
     } else if (iscomment(line,"%PDF-") && cmd_scan_pdf) {
       
-      /* PDF hack to set DELAYSAFER on interpreter invocation */
-      pdf_delaysafer_hack = 1;
-
       struct document *retval = NULL;
       FILE *tmpfile = (FILE*)NULL;
       char *filename_dsc;
