diff -ru gv-3.6.2/src/Ghostview.c gv-3.6.2J/src/Ghostview.c
--- gv-3.6.2/src/Ghostview.c	2006-07-04 13:09:52.000000000 -0600
+++ gv-3.6.2J/src/Ghostview.c	2007-04-18 16:12:54.000000000 -0600
@@ -1395,7 +1395,6 @@
 	 while (isspace(*dptr)) dptr++;
     }
     argv[argc++] = "-dNOPAUSE";
-    argv[argc++] = "-g2x2"; /* Avoid unwanted rotation of landscape pdf files */
     if (gvw->ghostview.quiet) argv[argc++] = "-dQUIET";
     if (gvw->ghostview.safer) 
       {
@@ -2016,6 +2015,7 @@
     GhostviewWidget gvw = (GhostviewWidget) w;
     BEGINMESSAGE(GhostviewIsInterpreterRunning)
     ENDMESSAGE(GhostviewIsInterpreterRunning)
+    if(gvw->ghostview.disable_start) return 0;
     return gvw->ghostview.interpreter_pid != -1;
 }
 
Only in gv-3.6.2J/src: Ghostview.c.orig
diff -ru gv-3.6.2/src/Makefile.am gv-3.6.2J/src/Makefile.am
--- gv-3.6.2/src/Makefile.am	2006-09-14 13:10:10.000000000 -0600
+++ gv-3.6.2J/src/Makefile.am	2007-04-18 16:12:26.000000000 -0600
@@ -195,7 +195,7 @@
 	@echo "GV.gsX11AlphaDevice:	-sDEVICE=x11 -dTextAlphaBits=4 -dGraphicsAlphaBits=2 -dMaxBitmap=10000000 -dNOPLATFONTS" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsSafer:		True" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.gsQuiet:		True" >> $(srcdir)/gv_intern_res_unix.dat
-	@echo "GV.gsArguments:" >> $(srcdir)/gv_intern_res_unix.dat
+	@echo "GV.gsArguments:		-dFIXEDMEDIA" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.uncompressCommand:	gzip -d -c %s > %s" >> $(srcdir)/gv_intern_res_unix.dat
 	@echo "GV.printCommand:	$(PRINT_COMMAND)" >> $(srcdir)/gv_intern_res_unix.dat
 
diff -ru gv-3.6.2/src/info.c gv-3.6.2J/src/info.c
--- gv-3.6.2/src/info.c	2004-11-11 05:21:05.000000000 -0700
+++ gv-3.6.2J/src/info.c	2007-04-18 16:12:49.000000000 -0600
@@ -76,6 +76,7 @@
    Widget	w;
    XtPointer	client_data, call_data;
 {
+   if(gv_gs_quiet) return;
    BEGINMESSAGE(cb_popupInfoPopup)
    if (!infoPopupCreated) { makeInfoPopup(); }
    if (infoPopupVisible==True) {INFMESSAGE(InfoPopup already up) ENDMESSAGE(popupInfoPopup) return; }
diff -ru gv-3.6.2/src/main.c gv-3.6.2J/src/main.c
--- gv-3.6.2/src/main.c	2006-09-14 12:56:08.000000000 -0600
+++ gv-3.6.2J/src/main.c	2007-04-18 16:12:49.000000000 -0600
@@ -594,7 +594,7 @@
      XrmDatabase db;
      INFMESSAGE(getting resources)
      db = resource_buildDatabase(gv_display,gv_class,gv_name,&argc,argv);
-     XrmCombineDatabase(db,&gv_database,False);
+     XrmCombineDatabase(db,&gv_database,True);
    }
 
 /*### initializing widget set and creating application shell #########################*/
@@ -1033,8 +1033,7 @@
             XtSetArg(args[n], XtNinterpreter,gv_gs_interpreter);	n++;
             b = gv_gs_safer ? True : False;
             XtSetArg(args[n], XtNsafer,b);                              n++;
-            b = gv_gs_quiet ? True : False;
-            XtSetArg(args[n], XtNquiet,b);                              n++;
+            XtSetArg(args[n], XtNquiet,True);                              n++;
             b = app_res.use_bpixmap ? True : False;
             XtSetArg(args[n], XtNuseBackingPixmap,b);                   n++;
             XtSetArg(args[n], XtNarguments,gv_gs_arguments);            n++;
diff -ru gv-3.6.2/src/ps.c gv-3.6.2J/src/ps.c
--- gv-3.6.2/src/ps.c	2006-07-07 09:35:49.000000000 -0600
+++ gv-3.6.2J/src/ps.c	2007-04-18 16:12:20.000000000 -0600
@@ -1396,6 +1396,8 @@
 	quoted=1;
 	line++;
 	while (*line && !(*line == ')' && level == 0 )) {
+	    if (cp - text >= PSLINELENGTH - 1)
+                break;
 	    if (*line == '\\') {
 		if (*(line+1) == 'n') {
 		    *cp++ = '\n';
@@ -1450,8 +1452,11 @@
 	    }
 	}
     } else {
-	while (*line && !(*line == ' ' || *line == '\t' || *line == '\n'))
+        while (*line && !(*line == ' ' || *line == '\t' || *line == '\n')) {
+            if (cp - text >= PSLINELENGTH - 2)
+                break;
 	    *cp++ = *line++;
+	}
     }
     *cp = '\0';
     if (next_char) *next_char = line;
