diff -ru gv-3.6.1/src/main.c gv-3.6.1J/src/main.c
--- gv-3.6.1/src/main.c	2004-12-06 16:36:57.000000000 -0700
+++ gv-3.6.1J/src/main.c	2005-02-23 22:27:05.625015080 -0700
@@ -421,7 +421,7 @@
    /*### Manage GNU command line arguments ########################*/
    int opt_counter = 0;
    int c;
-   while ((c = getopt_long (argc, argv, "vhu", GNU_longOptions, NULL))
+   while ((c = getopt_long_only (argc, argv, "vhu", GNU_longOptions, NULL))
 	  != -1)
      {
 
