diff -ru gv-3.6.1/src/gv_signal.c gv-3.6.1J/src/gv_signal.c
--- gv-3.6.1/src/gv_signal.c	2004-11-11 05:21:05.000000000 -0700
+++ gv-3.6.1J/src/gv_signal.c	2005-03-14 20:00:58.613998872 -0700
@@ -209,10 +209,10 @@
     sid_updateFile = sid_terminate = 0;
   }
   signal(SIGHUP,  shu);
-  signal(SIGINT,  sht);
   signal(SIGQUIT, sht);
   signal(SIGTERM, sht);
 #if 0
+  signal(SIGINT,  sht);
   signal(SIGILL,  sht);
   signal(SIGBUS,  sht);
   signal(SIGFPE,  sht);
