diff -ru ImageMagick-6.2.4/magick/animate.c ImageMagick-6.2.4J/magick/animate.c
--- ImageMagick-6.2.4/magick/animate.c	2005-08-12 10:27:48.000000000 -0600
+++ ImageMagick-6.2.4J/magick/animate.c	2005-08-15 19:53:31.000000000 -0600
@@ -1201,8 +1201,8 @@
             i++;
             if (i == (long) argc)
               ThrowAnimateException(OptionError,"MissingArgument",option);
-            if (IsGeometry(argv[i]) == MagickFalse)
-              ThrowAnimateInvalidArgumentException(option,argv[i]);
+//            if (IsGeometry(argv[i]) == MagickFalse)
+//              ThrowAnimateInvalidArgumentException(option,argv[i]);
             first_scene=atol(argv[i]);
             last_scene=first_scene;
             (void) sscanf(argv[i],"%ld-%ld",&first_scene,&last_scene);
