diff -ru pstoedit-3.43/src/drvasy.cpp pstoedit-3.43J/src/drvasy.cpp
--- pstoedit-3.43/src/drvasy.cpp	2006-03-04 14:16:02.000000000 -0700
+++ pstoedit-3.43J/src/drvasy.cpp	2006-03-07 02:50:40.000000000 -0700
@@ -62,7 +62,7 @@
 {
   // Output copyright information
   outf << "// Converted from PostScript(TM) to Asymptote by pstoedit\n"
-       << "// Asymptote 0.84 (or later) backend contributed by John Bowman\n"
+       << "// Asymptote 1.00 (or later) backend contributed by John Bowman\n"
        << "// pstoedit is Copyright (C) 1993 - 2006 Wolfgang Glunz"
        << " <wglunz34_AT_pstoedit.net>\n\n";
 	
@@ -286,7 +286,7 @@
   ostringstream buf;
   buf << outBaseName << "." << imgcount << ".eps";
   
-  outf << "label(include(\"" << buf.str() << "\"),("
+  outf << "label(graphic(\"" << buf.str() << "\"),("
        << ll.x_ << "," << ll.y_ << "),align);" << endl;
   outf << "layer();" << endl;
   
