diff -ru Imaging-1.1.6/Tk/tkImaging.c Imaging-1.1.6J/Tk/tkImaging.c
--- Imaging-1.1.6/Tk/tkImaging.c	2006-12-03 04:37:29.000000000 -0700
+++ Imaging-1.1.6J/Tk/tkImaging.c	2007-11-13 00:56:48.000000000 -0700
@@ -48,6 +48,9 @@
    for the Tcl_CreateCommand command. */
 #define USE_COMPAT_CONST
 
+#define CONST84 const
+#define CONST84_RETURN const
+
 #include "tk.h"
 
 #include "Imaging.h"
@@ -146,9 +149,6 @@
         int x, y;
         Tk_PhotoImageBlock run;
 
-        /* Clear current contents */
-        Tk_PhotoBlank(photo);
-
         /* Setup run descriptor */
         run.height = 1;
         run.pitch = block.pitch;
@@ -156,7 +156,7 @@
         run.offset[0] = 0;
         run.offset[1] = 1;
         run.offset[2] = 2;
-	run.offset[3] = 0; /* no alpha (or reserved, under 8.2) */
+	run.offset[3] = 3; /* alpha (or reserved, under 8.2) */
 
         /* Copy opaque runs to photo image */
         for (y = 0; y < block.height; y++) {
diff -ru Imaging-1.1.6/_imagingtk.c Imaging-1.1.6J/_imagingtk.c
--- Imaging-1.1.6/_imagingtk.c	2006-12-03 04:51:25.000000000 -0700
+++ Imaging-1.1.6J/_imagingtk.c	2007-11-13 00:56:02.000000000 -0700
@@ -17,6 +17,9 @@
 #include "Python.h"
 #include "Imaging.h"
 
+#define CONST84 const
+#define CONST84_RETURN const
+
 #include "tk.h"
 
 /* must link with Tk/tkImaging.c */
